### PR TITLE
Intra feature MS2 similarity

### PR DESCRIPTION
**Pull request by robinschmid**
This can be used to estimate alignment quality and MS2 similarity across samples.

image

Can also be exported for each value
image

### CODE REVIEW

**Comment by robinschmid** [comment]
changed format to scores formta

File: src/main/java/io/github/mzmine/modules/dataanalysis/feat_ms2_similarity_intra/IntraFeatureMs2SimilarityParameters.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      new FileChooser.ExtensionFilter("comma-separated values", "*.csv"), //
+      new FileChooser.ExtensionFilter("All files", "*.*") //
+  );
+
+  public static final OptionalParameter<FileNameParameter> filename = new OptionalParameter<>(
+      new FileNameParameter("Export to file", "Name of the output CSV file.", extensions,
+          FileSelectionType.SAVE), false);
+
+  public static final MZToleranceParameter mzTol = new MZToleranceParameter("m/z tolerance (MS2)",
+      "Tolerance value of the m/z difference between MS2 signals (add absolute tolerance to cover small neutral losses (5 ppm on m=18 may be insufficient))",
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataanalysis/feat_ms2_similarity_intra/IntraFeatureMs2SimilarityParameters.java:54]: m=? i think it's supposed to be m/z?

File: src/main/java/io/github/mzmine/modules/dataanalysis/feat_ms2_similarity_intra/IntraFeatureMs2SimilarityTask.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class IntraFeatureMs2SimilarityTask extends AbstractTask {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataanalysis/feat_ms2_similarity_intra/IntraFeatureMs2SimilarityTask.java:63]: It's an intra-row similarity, so let's call it that.

File: src/main/java/io/github/mzmine/gui/mainwindow/MainMenu.fxml [hunk 3]
```diff
@@ -352,6 +352,8 @@
         userData="io.github.mzmine.modules.dataprocessing.filter_groupms2_refine.GroupedMs2RefinementModule"/>
       <MenuItem onAction="#runModule" text="Precursor purity checker (chimerity)"
         userData="io.github.mzmine.modules.dataanalysis.spec_chimeric_precursor.PrecursorPurityCheckerModule"/>
+      <MenuItem onAction="#runModule" text="Score intra feature purity by MS2 similarity"
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/gui/mainwindow/MainMenu.fxml:355]: row

**Review by SteffenHeu — CHANGES_REQUESTED** [review]
Change the naming to row to make it less confusing please

File: src/main/java/io/github/mzmine/gui/mainwindow/MainMenu.fxml (hunk 3 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/gui/mainwindow/MainMenu.fxml:355]: done

**Comment by robinschmid** [comment]
done. Maybe we can wait for your fix to be merged

**Pull request merged**